File: content/problems/0763-partition-labels/solutions/last-occurrence-greedy/Solution.java
class Solution {
    public List<Integer> partitionLabels(String s) {
        int[] last = new int[26]; for (int index = 0; index < s.length(); index++) last[s.charAt(index) - 'a'] = index;
        List<Integer> answer = new ArrayList<>(); int start = 0, end = 0;
        for (int index = 0; index < s.length(); index++) { end = Math.max(end, last[s.charAt(index) - 'a']); if (index == end) { answer.add(end - start + 1); start = index + 1; } }
        return answer;
    }
}

File: content/problems/0763-partition-labels/solutions/last-occurrence-greedy/solution.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
        array<int, 26> last{}; for (int index = 0; index < (int)s.size(); ++index) last[s[index] - 'a'] = index;
        vector<int> answer; int start = 0, end = 0;
        for (int index = 0; index < (int)s.size(); ++index) { end = max(end, last[s[index] - 'a']); if (index == end) { answer.push_back(end - start + 1); start = index + 1; } }
        return answer;
    }
};
